### DISCUSSION TITLE
{gsm} v2 feedback thread.

### DISCUSSION THREAD

**Discussion by jwildfire** [Gilead-Public/gsm#1590] [General]
Feedback on running `\inst\examples` or any other aspects of Gismo v2

**Comment by william-lee-96** [comment]
Hi @jwildfire, I wanted to ask a few gsm-related questions that came out of our CSM group:

Regarding the 12 standard KRIs in the HTML report:
- In slide 2 of this [presentation](https://drive.google.com/file/d/1Dlu8CoGF4TBCQwOQM75VJtGqTohsQ15F/view), we can see that for kri0001, the "Analysis Set" is filtered to just enrolled subjects. Is there documentation on which "Analysis Set" is being used for each KRI?
- Secondarily, are there plans in v2 to add an "Analysis Set" column in the HTML report KRI Glossary, where the value for kri0001 would be something like "Enrolled"? This is something we'd be very interested in! (Alternatively, we could see this info being added to the "Data Filters" column)
- Days on Study is defined as "Enroll to d/c date" -- does "d/c date" mean "discontinuation/completion date"?
- In v1, when we use a different stat method (e.g., Fisher) in one of the KRI workflows, this change is not reflected in the KRI Glossary (the model still says "Normal Approximation") -- will this be changed in v2?

And regarding what gets sent to the TMF:
- Are you just sending the HTML report as is?

Thank you! I'm going to be running/exploring the v2 code today, and will drop my feedback in this thread shortly.

**Reply by jwildfire** [reply]
Hi @william-lee-96 - Thanks for the notes. Initial replies below. We're still grinding away on v2, so I might hold off on some of those questions until next week. Here are initial thoughts on the v1 stuff though.


> Regarding the 12 standard KRIs in the HTML report:
> 
> * In slide 2 of this [presentation](https://drive.google.com/file/d/1Dlu8CoGF4TBCQwOQM75VJtGqTohsQ15F/view), we can see that for kri0001, the "Analysis Set" is filtered to just enrolled subjects. Is there documentation on which "Analysis Set" is being used for each KRI?

It's embedded in the workflows (see [here](https://github.com/Gilead-BioStats/gsm/blob/aafc36eef839131de176cbb9c405b2ad7184c83b/inst/workflow/kri0001.yaml#L2-L8)) but not well documented. First 11 KRIs are typically run on enrolled participants and 12th KRI (screen failures) is on screened. Can be customized for a given study by updating workflow if needed. 

> * Secondarily, are there plans in v2 to add an "Analysis Set" column in the HTML report KRI Glossary, where the value for kri0001 would be something like "Enrolled"? This is something we'd be very interested in! (Alternatively, we could see this info being added to the "Data Filters" column)

V2 workflows have a [meta header](https://github.com/Gilead-BioStats/gsm/blob/918e9e2c9b63213bfd0e1f543f94b96fb481513a/inst/workflow/metrics/kri0001.yaml#L1-L12) where a "population" or "AnalysisSet" parameter could easily be added. We don't have it currently, but I think it's a good idea. Feel free to file an issue (or a PR!) adding. 

> * Days on Study is defined as "Enroll to d/c date" -- does "d/c date" mean "discontinuation/completion date"?

I think we were just using "d/c" for discontinuation, but yes, it's days from enrollment to discontinuation or completion (or to the snapshot date for active participants). I'd imagine the exact definition could vary a bit between organizations which would be fine. Whatever value is provided in the data will be used. 

> * In v1, when we use a different stat method (e.g., Fisher) in one of the KRI workflows, this change is not reflected in the KRI Glossary (the model still says "Normal Approximation") -- will this be changed in v2?

Yes, we have a `model` parameter in the `meta` header mentioned above that gets surfaced in the report. 

> And regarding what gets sent to the TMF:
> 
> * Are you just sending the HTML report as is?

We're currently filing an adjudicated summary of findings instead of the report itself, but process could vary between orgs. I don't think it's unreasonable to file the whole report.

**Comment by william-lee-96** [comment]
Hi dev team, I am running the code in `gsm/inst/examples/workflow_charts.R`, and am running into a problem with the call to the function `Visualize_Metric` (lines 49-55). Here is the error message I am getting:

```
Error in Visualize_Metric(dfSummary = dfSummary, dfBounds = dfBounds,  : 
  unused argument (dfSite = clindata::ctms_site %>% rename(SiteID = site_num))
```

Looking at the function's source code (in `gsm/R/Visualize_Metric.R`), I can see that this `dfSite` parameter should instead be `dfGroups` -- however, after making this change, I then get this error when running lines 49-55:

`Error in strsplit(strThreshold, ",") : non-character argument
`

Any ideas on what might be going wrong here? All the code up to line 49 runs without issue.

**Reply by jwildfire** [reply]
Yeah, we've gone back and forth on this a bit and are actively refactoring this process still. I think we've got the solution lined up and just need to finish implementing. Will get examples updated once we've merged in to dev and ping you to re-test.

**Comment by william-lee-96** [comment]
Just flagging that -- with the exception of a  a single space on line 63 -- `examples/workflow_charts.R` and `examples/workflow_report.R` are the same script

**Comment by william-lee-96** [comment]
I'm working my way through `examples/workflow_reporting.R`, but when I try to run line 24, I get this error:

```
Error in RunWorkflows(lWorkflow = wf_kri, lData = lMapped) : 
  could not find function "RunWorkflows"
```

I can see that the source code for this functions exists in `gsm/R/RunWorkflows.R`, so not sure what the issue might be -- please let me know if you have any ideas!

**Reply by jwildfire** [reply]
hmm. I'm not sure either. Let's try again once we have the releases candidate and then we can debug if this is still popping up.

**Comment by jonthegeek** [comment]
It bugs me that `Numerator`, `Denominator`, `Metric`, and `Score` have different meanings in `dfMetrics` than in all the other tables. If we want to join `dfMetrics` into `dfSummary` (to add metric `Abbreviation`s, for example), we need to explicitly exclude those columns to avoid column renames. Is there a reason we didn't go with NumeratorSource`, `DenominatorSource`, `MetricName`, and `ScoreType`?

I don't know if there are discussions about this elsewhere, nor how much of a pain it would be to change at this point, but it's something I've noticed in the last couple days as I really start to work with the data.

**Comment by ianamovsesian** [comment]
Hi @jwildfire, @lauramaxwell 

I see there is now a new set of examples in dev/inst/examples
Should we disregard the old example set then and only focus on new ones?
